baseline-error-prone/src/main/java/com/palantir/baseline/errorprone/UnnecessaryLambdaArgumentParentheses.java

```diff
@@ -0,0 +1,97 @@
+/*
+ * (c) Copyright 2020 Palantir Technologies Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.palantir.baseline.errorprone;
+
+import com.google.auto.service.AutoService;
+import com.google.errorprone.BugPattern;
+import com.google.errorprone.VisitorState;
+import com.google.errorprone.bugpatterns.BugChecker;
+import com.google.errorprone.fixes.SuggestedFix;
+import com.google.errorprone.matchers.Description;
+import com.google.errorprone.util.ErrorProneToken;
+import com.sun.source.tree.LambdaExpressionTree;
+import com.sun.tools.javac.parser.Tokens;
+import com.sun.tools.javac.tree.JCTree;
+import java.util.List;
+
+/**
+ * UnnecessaryLambdaArgumentParentheses provides similar functionality to the upstream UnnecessaryParentheses, but
+ * specifically for single-parameter lambda arguments which are not covered by the existing check. Perhaps this can be
+ * contributed upstream. There's an argument against combining the two because parentheses around lambda arguments
+ * cannot be parsed directly from the AST where other parenthesis checked by UnnecessaryParentheses can.
+ */
+@AutoService(BugChecker.class)
+@BugPattern(
+        name = "UnnecessaryLambdaArgumentParentheses",
+        link = "https://github.com/palantir/gradle-baseline#baseline-error-prone-checks",
+        linkType = BugPattern.LinkType.CUSTOM,
+        providesFix = BugPattern.ProvidesFix.REQUIRES_HUMAN_ATTENTION,
+        severity = BugPattern.SeverityLevel.WARNING,
```

Comment:
I think we should go with error since checkstyle previous enforced this

Reply:
Most projects fail on WARNING -- iirc we tried using ERROR for the `BracesRequired` check but ran into issues (possibly with generated code).

Since excavator will auto-fix anything in projects that don't fail on warnings, I don't think there's much risk.

Reply:
ok sure!